httpbakery/form/form.go

```diff
@@ -58,27 +58,44 @@ type Filler interface {
 
 // SetUpAuth configures form authentication on c. The VisitWebPage field
 // in c will be set to a function that will attempt form based
-// authentication using h to perform the interaction with the user.
+// authentication using f to perform the interaction with the user.
 //
-// The VisitWebPage function downloads the schema from the specified
-// server and calls h.Handle. The map returned by h.Handle should match
-// the schema specified, but VisitWebPage will not verify this before
-// sending to the server. Any errors returned by h.Handle will not have
-// their cause masked.
+// c.VisitWebPage will be set to a function created by calling
+// VisitWebPage using c as the client, f as the filler and the current
+// value of c.VisitWebPage for fallback.
 func SetUpAuth(c *httpbakery.Client, f Filler) {
-	v := webPageVisitor{
-		client: &httprequest.Client{
+	c.VisitWebPage = VisitWebPage(
+		&httprequest.Client{
 			Doer: c,
 		},
-		filler: f,
+		f,
+		c.VisitWebPage,
+	)
+}
+
+// VisitWebPage creates a function suitable for use with
+// httpbakery.Client.VisitWebPage. The new function downloads the schema
+// from the specified server and calls f.Fill. The map returned by f.Fill
+// should match the schema specified, but this is not verified before
+// sending to the server. Any errors returned by f.Fill will not have
+// their cause masked.
+//
+// If the new function detects that form login is not supported by the
+// server and fb is not nill then fb will be called to perform the visit.
+func VisitWebPage(c *httprequest.Client, f Filler, fb func(u *url.URL) error) func(u *url.URL) error {
```

Comment:
I think I'd probably call it fallback not fb to emphasise the relationship with webPageVisitor.fallback. I don't mind much though.